src/components/WalletBalanceList/index.tsx

```diff
@@ -26,157 +27,184 @@ const ASSET_DEFAULT_ICON_URL = `${process.env
   wallet,
   isKycPassed,
-  assetsAvailableForDeposit = []
-}) => (
-  <div className="wallet__balances">
-    {wallet.hasBalances || (
-      <small style={{margin: 0}}>You don’t have any asset yet</small>
-    )}
-    {wallet.hasBalances &&
-      Object.keys(wallet.getBalancesByCategory).map(x => {
-        const balances = wallet.getBalancesByCategory[x];
-        return (
-          <div key={x}>
-            <h3>
-              {x}
-              <small>
-                {balances.length} {plural(balances.length, 'asset')}
-              </small>
-            </h3>
-            <table className="table_assets">
-              <thead>
-                <tr>
-                  <th className="_asset">
-                    <span>Asset</span>
-                  </th>
-                  <th className="_currency">Base currency</th>
-                  <th className="_amount">Amount</th>
-                  <th className="_action">&nbsp;</th>
-                </tr>
-              </thead>
-              <tbody>
-                {balances.map(b => (
-                  <tr key={b.assetId + b.balance}>
-                    <td className="_asset">
-                      <div className="issuer">
-                        <div className="issuer__img">
-                          <img
-                            src={b.asset.iconUrl || ASSET_DEFAULT_ICON_URL}
-                            alt="asset"
-                            width={48}
-                            height={48}
-                          />
-                        </div>
-                        <div className="issuer__content">
-                          <div className="issuer__name">
-                            {wallet.isTrading ? (
-                              <Link to={ROUTE_ASSET(b.assetId)}>
-                                {b.asset.name}
-                              </Link>
-                            ) : (
-                              b.asset.name
-                            )}
+  assetsAvailableForCreditCardDeposit = [],
+  assetsAvailableForSwiftDeposit = []
+}) => {
+  const isAvailableForCreditCardDeposit = (assetId: string) =>
+    assetsAvailableForCreditCardDeposit.find(asset => asset.id === assetId);
+  const isAvailableForSwiftDeposit = (assetId: string) =>
+    assetsAvailableForSwiftDeposit.find(asset => asset.id === assetId);
+
+  const renderDepositMenuItem = (
+    label: string,
+    route: string,
+    iconUrl: string
+  ) => (
+    <DropdownListItem
+      key={label}
+      className={classnames({
+        disabled: !isKycPassed
+      })}
+    >
+      {isKycPassed ? (
+        <Link to={route}>
+          <img className="icon" src={iconUrl} />
+          {label}
+        </Link>
+      ) : (
+        <a>
+          <img className="icon" src={iconUrl} />
+          {label}
+        </a>
+      )}
+    </DropdownListItem>
+  );
+
+  return (
+    <div className="wallet__balances">
+      {wallet.hasBalances || (
```

Comment:
I may be wrong because component is too big, but you can return null at the beginning of the function if !wallet.hasBalances is true.

Reply:
I would also recommend you split this component to a smaller components or render functions.

Reply:
Yeap agree, but probably not in scope of this task